typings/index.d.ts

```diff
@@ -14,11 +14,85 @@ declare module 'discord.js' {
   import { ChildProcess } from 'child_process';
   import { EventEmitter } from 'events';
   import { PathLike } from 'fs';
+  import { Response } from 'node-fetch';
   import { Readable, Stream, Writable } from 'stream';
+  import { URLSearchParams } from 'url';
   import * as WebSocket from 'ws';
 
   export const version: string;
 
+  //#region Rest
+
+  class AsyncQueue {
+    // tslint:disable-next-line:ban-types
+    private readonly promises: { promise: Promise<void>; resolve: Function }[];
+    public remaining: number;
+    public wait(): Promise<void>;
+    private shift(): void;
+  }
+
+  class RequestHandler {
+    private readonly manager: RESTManager;
+    private readonly queue: AsyncQueue;
+    public reset: number;
+    public remaining: number;
+    public limit: number;
+    public retryAfter: number;
+    public limited: boolean;
+    private readonly inactive: boolean;
+
+    private execute(): Collection<string, unknown> | Buffer;
+    public push(): Collection<string, unknown> | Buffer;
+  }
+
+  interface RequestOptions {
+    query?: URLSearchParams | Record<string, string | string[]>;
+    versioned?: boolean;
+    auth?: boolean;
+    reason?: string;
+    headers?: Record<string, string>;
+    data?: Record<string, unknown>;
```

Comment:
Those are very troublesome, for example:

```typescript
interface Headers {
	Authorization: string;
}

const headers: Headers = { Authorization: 'Hello World' };

const options: RequestOptions = { headers };
```

Causes this error:

```typescript
Type 'Headers' is not assignable to type 'Record<string, string>'.
  Index signature is missing in type 'Headers'.(2322)
```

To properly fix it, the `{}` type (albeit discouraged by a TS ESlint rule) accepts *any* non-null object, which is what we want.

Reply:
Resolved in https://github.com/discordjs/discord.js/pull/5254/commits/e5ce43d328229f40dce8d92af40b21596ae50d98